classification/feature_generation/fishing_feature_generation.py

```diff
@@ -0,0 +1,138 @@
+import calendar
+import numpy as np
+import tensorflow as tf
+from . import feature_generation
+from . import feature_utilities
+
+def input_fn(vessel_metadata,
+             filenames,
+            num_features,
+            max_time_delta,
+            window_size,
+            min_timeslice_size,
+            parallelism=4,
+            num_slices_per_id=8):
+
+    random_state = np.random.RandomState()
+
+    
+    def xform(id_, movement_features):
+
+        def _xform(features, int_id):
+            # Extract several random windows from each vessel track
+            # TODO: Fix feature generation so it returns strings directly
```

Comment:
Should we solve the TODO?

Reply:
@enriquetuya : that involves enough changes that I'm reluctant to do that now.  My tentative plan is fix this on the bird life style feature generation I putting together for loitering.  Then start moving models over to that pipeline.

Reply:
Bird life style feature generation? Now I'm intrigued...

Reply:
@d40cht we put together a model for bird life that predicts longline sets (setting only, ignores hauling), that works quite well. It used time points interpolated onto a constant grid (and fewer features, although that's not the key point).  We'd like to move to interpolating onto a constant grid in the nearish future.